ntp-proto/src/tls_utils.rs

```diff
@@ -0,0 +1,22 @@
+pub struct AllowAnyAnonymousOrCertificateBearingClient;
```

Comment:
why not `AllowAnyClient`?

Reply:
in other words, what makes this different from `NoClientAuth`. I'd like some comments for why this is needed/what the goal is.

Reply:
Check on the "needs more comments part".

- `AllowAnyClient` feels too imprecise; the RusTLS built-in verifier that is closest is called `AllowAnyAnonymousOrAuthenticatedClient`, which also sounds like "allow any client"; except that it will refuse a connection if a certificate is presented that is not validated against a trust root. The intent of this verifier is that it will accept any anonymous connection (i.e. no client certificate) as well as those with a well-formed certificate (i.e. clients bearing a certificate), which is not the same as being "authenticated". (I'm not wedded to the ridiculously long name `AnyAnonymousOrCertificateBearingClient`, but it has the benefit of being very closely related to the name RusTLS chose).

- `NoClientAuth` is the same as `no_client_auth`, i.e. client certificates don't end up in `peer_certificates()` and that means we can't check whether someone is allowed to send `FixedKeyRequests`.

Reply:
I approve of this name, consistent with rustls which i think is the most important here.